wasm_playground/maps/src/main.rs

```diff
@@ -0,0 +1,964 @@
+}
+#[no_mangle]
+pub fn get_i32_i32(name: i32, key: i32) -> i32 {
+    get_i32(name, &key)
+}
+#[no_mangle]
+pub fn insert_i32_string(name: i32, key: i32, offset: i32, length: i32){
+    let value = string_from_data(offset, length);
+    if ! insert_i32_string_inner(name, key, value){
+        panic!("Failed to insert into i32_string map");
+    }
+}
+#[no_mangle]
+pub fn get_string_from_i32string(name: i32, key: i32) -> String {
+    get_string(name, &key)
+}
+#[no_mangle]
+pub fn string_from_data(offset: i32, length: i32) -> String {
+    let callee_ptr: *const u8 = offset as *const u8;
+    let callee_slice: &[u8] = unsafe { slice::from_raw_parts(callee_ptr, usize::try_from(length).unwrap()) }; 
+    String::from_utf8(callee_slice.to_vec()).unwrap()
+}
+//This is the stuff for report var 
+
+// #[no_mangle]
+// pub fn add_report_map(name: i32) {
+//     REPORT_MAPS.lock().unwrap().push(name);
+// }
+// #[no_mangle]
+// pub fn output_report_maps() {
+//     for name in REPORT_MAPS.lock().unwrap().iter() {
+//         print_map(*name);
+//     }
+//     println!();
+// }
+// #[no_mangle]
+// pub fn print_info(gid: i32, val: i32){
+//     println!("GID: {} -> {}", gid, val);
+// }
+#[no_mangle]
+pub fn print_map(map_id: i32) -> String {
+    let binding = MY_MAPS.lock().unwrap();
+    let map = binding.get(&map_id).unwrap();
+    format!("{}\n", map.dump_map())
+}
+#[no_mangle]
+pub fn print_meta() {
+    let mut running_output: String = String::new();
+    let binding = MY_MAPS.lock().unwrap();
+    let var_meta = binding.get(&0).expect("No metadata for variables found");
+    let map_meta = binding.get(&1).expect("No metadata for maps found");
+    match var_meta {
+        AnyMap::i32_string_Map(map) => {
+            for (key, value) in map.iter() {
+                running_output.push_str(&format!("GID: {}\t{}\t", key, value));
+                let map = binding.get(&key).expect("Metadata but no map found");
+                running_output.push_str(&format!("{}\n", map.dump_map()));
+            }
+        }
+        _ => {
+            panic!("Invalid metadata for variables");
+        }
+    }
+    match map_meta {
+        AnyMap::i32_string_Map(map) => {
+            for (key, value) in map.iter() {
+                running_output.push_str(&format!("MapId: {}\t{}\t", key, value));
+                let map = binding.get(&key).expect("Metadata but no map found");
+                running_output.push_str(&format!("{}\n", map.dump_map()));
+            }
+        }
+        _ => {
+            panic!("Invalid metadata for maps");
+        }
+    }
+    println!("{}", running_output);
+}
+
+
+//MAIN STARTS HERE
+#[no_mangle]
+pub fn foo(a: i32) -> i32 {
+    return inner_fn(a);
+}
+
+#[no_mangle]
+pub fn bar(a: i32) -> i32 {
+    return inner_fn(a);
+}
+
+#[no_mangle]
+pub fn inner_fn(a: i32) -> i32 {
+    return a*15 - 3;  
+}
+// #[no_mangle]
+// pub extern "C" fn _start() {
+//     main();
+// }
```

Comment:
Do you need this?

Reply:
cargo fmt does it to some of them and not others